### ISSUE TITLE
Restriction sites containing ambiguous bases do not digest targets 

### ISSUE THREAD

**Issue by blthree** [klavinslab/coral#31]
When trying to digest the vector pENTR223.1 with an SfiI RestrictionSite object from the Rebase.get() method, I noticed that it always returns a single uncut DNA object, when it should cut twice and return two sequences. I tried a few things and it seems like the issue is that the SfiI recognition sequence contains 5 'N' bases, GGCCNNNNNGGCC. I manually created two restriction site objects with the ambiguous N's replaced with the corresponding vector sequences, which worked fine. I also tried the reverse, N-masking the vector sequence at the cut site, which caused weird behavior with the rebase restriction site. That digest output the 2 correct fragments, as well as two empty DNA objects. Below is the script I used for testing.

```
import coral as cor


# fragment of pENTR223.1
# original seq and N-masked version
vector_seq = 'cccaaataatgattttattttgactgatagtgacctgttcgttgcaacaaattgatgagcaatgcttttttataatgccaactttgtacaaaaaagcaga' \
                  'agggccgtcaaggccagaaggagatataaccatgtaaggcctcatgggcccagctttcttgtacaaagttggcattataaaaaataattgctcatcaatt'
vector_seq_NNNNN = 'cccaaataatgattttattttgactgatagtgacctgttcgttgcaacaaattgatgagcaatgcttttttataatgccaactttgtacaaaaaagcaga' \
                  'agggccNNNNNggccagaaggagatataaccatgtaaggccNNNNNggcccagctttcttgtacaaagttggcattataaaaaataattgctcatcaatt'

vector = cor.DNA(vector_seq, circular=True) # DNA object from original vector sequence
vector_NNNNN = cor.DNA(vector_seq_NNNNN, circular=True) # DNA object from vector sequence with NNNNN to match SfiI cut site

# define restriction sites
# from rebase 
reb = cor.database.Rebase()
sfiI = reb.get('SfiI')
#  make two manually defined sfiI restriction sites
sfiIa = cor.sequence.RestrictionSite(cor.DNA('GGCCgtcaaGGCC'), (8, 5) )
sfiIb = cor.sequence.RestrictionSite(cor.DNA('GGCCtcatgGGCC'), (8, 5) )

# digest original vector with sfiI from rebase
# yields one fragment... does not cut!

digest_one = cor.reaction.digest(vector.copy(), sfiI)
print('Original vector with rebase sfiI digest yields ' + str(len(digest_one)) + ' fragments')

# digest N-masked vector with sfiI from rebase
# yields 4 fragments... 2 empty objects and 2 correct objects

digest_two = cor.reaction.digest(vector_NNNNN.copy(), sfiI)
print('N-masked vector with rebase sfiI digest yields ' + str(len(digest_two)) + ' fragments')
for seq in digest_two:
    print(len(seq))

# digest original vector with manually defined sfiI sites
# yields two fragments... correct!

digest_three = cor.reaction.digest(vector.copy(), sfiIa)
digest_three = cor.reaction.digest(digest_three[0].copy(), sfiIb)
print('Original vector with manually defined sfiI digest yields ' + str(len(digest_three)) + ' fragments')

# digest N-masked vector with manually defined sfiI sites
# yields one fragment... does not cut!

digest_four = cor.reaction.digest(vector_NNNNN.copy(), sfiIa)
digest_four = cor.reaction.digest(digest_three[0].copy(), sfiIb)
print('N-masked vector with manually defined sfiI digest yields ' + str(len(digest_four)) + ' fragments')

```

**Comment by blthree** [comment]
I did some quick fixes which worked for this case. See pull request #32

**Comment by nbolten** [comment]
Hi @blthree! Thanks for the bug report + pull request. Also, sorry for taking so long - I normally get to these sooner.

I changed the base branch of the request to `develop`, since that's where the next release is coming from. Once the tests pass, I'll merge (and add more tests)!

I walked through the example code and got this pattern of fragment numbers:
* 2
* 2
* 2
* 1

Let me know if this isn't the expected behavior!
